src/dmd/mars.d

```diff
@@ -2383,3 +2469,47 @@ Modules createModules(ref Strings files, ref Strings libmodules)
     }
     return modules;
 }
+
+/**
+Prints a supplied usage text to the console and
+returns the exit code for the help usage page.
+
+Params:
+    help = usage text to be printed
+
+Returns:
+    `EXIT_SUCCESS` if no errors occurred, `EXIT_FAILURE` otherwise
+*/
+byte printHelpUsage(string help)
+{
+    printf("%.*s", help.length, &help[0]);
+    return global.errors ? EXIT_FAILURE : EXIT_SUCCESS;
+}
+
+
+/**
+Checks whether a flag has no options (e.g. -foo)
+
+Params:
+    flag = pointer to the end of the flag
+
+Returns: `true` if the flags has no options
+*/
+bool hasNoOption(const(char)* p)
+{
+    return *p == 0;
+}
+
+/*
+Checks whether the option pointer supplied is a request
+for the help page.
+
+Params:
+    flag = pointer to the end of the flag
+
+Returns: `true` if `p` points to a help option
+*/
+bool isHelpOption(const(char*) flag)
+{
+    return strcmp(flag, "?") == 0;
```

Comment:
https://github.com/dlang/dmd/pull/9205 also comes with this which I already used this function here.

Reply:
I'm not sure en-functioning trivial one liners is justified.